pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -41,7 +41,7 @@ let
         (whenBetween "5.2" "5.18" yes)
       ];
       DEBUG_INFO_DWARF_TOOLCHAIN_DEFAULT = whenAtLeast "5.18" yes;
-      DEBUG_INFO_BTF            = whenAtLeast "5.2" (option yes);
+      DEBUG_INFO_BTF            = whenAtLeast "5.2" (option (if stdenv.hostPlatform.is32bit && (versionAtLeast version "5.15") then no else yes));
```

Comment:
We should add a comment here explaining why 32 bit has it disabled

Reply:
> We should add a comment here explaining why 32 bit has it disabled

Done!